src/frontend/screens/Settings/ProjectSettings/YourTeam/index.tsx

```diff
@@ -62,43 +67,65 @@ export const YourTeam: NativeNavigationComponent<'YourTeam'> = ({
   const {formatMessage: t} = useIntl();
   const membersQuery = useProjectMembers();
   const deviceInfo = useDeviceInfo();
+  const projectSettings = useProjectSettings();
+
   const coordinators = !membersQuery.data
     ? []
     : membersQuery.data.filter(
         member =>
-          member.role.roleId === CREATOR_ROLE_ID ||
-          member.role.roleId === COORDINATOR_ROLE_ID,
+          member.role.roleId === COORDINATOR_ROLE_ID ||
+          member.role.roleId === CREATOR_ROLE_ID,
       );
 
   const participants = !membersQuery.data
     ? []
     : membersQuery.data.filter(member => member.role.roleId === MEMBER_ROLE_ID);
 
+  if (projectSettings.isLoading) {
+    return (
+      <CenteredView>
+        <Loading />
+      </CenteredView>
+    );
+  }
+
+  if (projectSettings.data && !projectSettings.data.name) {
+    return <NotOnProject />;
+  }
+
   return (
     <ScrollView style={styles.container}>
-      <Button
-        fullWidth
-        variant="outlined"
-        onPress={() => {
-          navigation.navigate('SelectDevice');
-        }}>
-        <View
-          style={{
-            flexDirection: 'row',
-            justifyContent: 'center',
-            alignItems: 'center',
+      {deviceInfo.isLoading ? (
+        <UIActivityIndicator size={30} />
+      ) : !deviceInfo.data ||
+        !coordinators.some(
+          coordinator => coordinator.deviceId === deviceInfo.data.deviceId,
+        ) ? null : (
```

Comment:
think it's worth extracting this into a variable e.g. `const shouldShowAddDeviceButton = ...`